### PR TITLE
Tanim

### PR DESCRIPTION
**Pull request by mmpozulp**
Let me know if you want me to split out into different PRs. The utility of having this in one PR is that it illustrates what I am trying to do. This PR adds new packages r-ks and r-multicool, adds a jdk version as per #8877, and widens the version constraint for some r packages.

### CODE REVIEW

**Comment by mmpozulp** [comment]
Here's my use case. I want to run
```python
from rpy2.robjects.packages import importr
print(importr("ks"))
```
after installing the requisite packages using spack. Here's how I did it

```console
ssh rztopaz

# clone spack from gh
git clone git@github.com:mmpozulp/spack.git lalor_spack4
cd lalor_spack4
git checkout tanim

# build
salloc -N1 --exclusive -t 2:00:00
./bin/spack install r-ks ^r@3.5.1 ^jdk@1.8.0_181-b13  
./bin/spack install py-rpy2@2.9.4 ^python@3.6.2 ^r@3.5.1 ^jdk@1.8.0_181-b13  
./bin/spack activate r-ks
./bin/spack activate py-rpy2
PATH=`./bin/spack location -i r@3.5.1`/bin:$PATH `./bin/spack location -i python@3.6.2`/bin/python3 -c 'from rpy2.robjects.packages import importr; print(importr("ks"))'
```

File: var/spack/repos/builtin/packages/r-ks/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    version('1.11.2', sha256='9dfd485096e1e67abc7dfcb7b76a83de110dd15bcfeffe5c899605b3a5592961')
+
+    depends_on('r@2.10:')
+
+    depends_on('r-fnn@1.1:')
+    depends_on('r-kernlab')
+    depends_on('r-mclust', type=('build', 'run'))
+    depends_on('r-multicool')
+    depends_on('r-mvtnorm@1.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-ks/package.py]: Missing a few dependencies:

`r-kernsmooth@2.22:`
`r-matrix`
`r-mgcv`

Also, all of these should be `type=('build', 'run')`, including `r`.

File: var/spack/repos/builtin/packages/r-lars/package.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ class RLars(RPackage):
     url      = "https://cran.r-project.org/src/contrib/lars_1.2.tar.gz"
     list_url = "https://cran.rstudio.com/src/contrib/Archive/lars"
 
-    depends_on('r@3.4.0:3.4.9')
+    depends_on('r@2.10:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-lars/package.py]: I know you didn't write this package, but can you move this dependency below the versions, and add `type=('build', 'run')`?

File: var/spack/repos/builtin/packages/r-multicool/package.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+class RMulticool(RPackage):
+    """Permutations of multisets in cool-lex order."""
+
+    homepage = "https://cran.r-project.org/package=multicool"
+    url      = "https://cran.r-project.org/src/contrib/Archive/multicool/multicool_0.1-9.tar.gz"
+
+    version('0.1-9', sha256='bdf92571cef1b649952d155395a92b8683099ee13114f73a9d41fc5d7d49d329')
+
+    depends_on('r')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-multicool/package.py]: Not needed, this is already present in `RPackage`.

File: var/spack/repos/builtin/packages/r-multicool/package.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    depends_on('r-rcpp@0.11.2:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-multicool/package.py]: `type=('build', 'run')`.

File: var/spack/repos/builtin/packages/r-pcapp/package.py [hunk 5]
```diff
@@ -38,5 +38,5 @@ class RPcapp(RPackage):
         version('1.9-60', '23dd468abb9fedc11e40166446df1017')
         version('1.9-50', 'be44f173404fd6e86ba0a5515711bfa3')
 
-        depends_on('r@3.4.0:3.4.9')
+        depends_on('r@3.4.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-pcapp/package.py]: According to https://cran.r-project.org/web/packages/pcaPP/index.html, there aren't any restrictions on the R version, so you can remove this.

File: var/spack/repos/builtin/packages/r-som/package.py [hunk 6]
```diff
@@ -38,4 +38,4 @@ class RSom(RPackage):
     version('0.3-3', 'd4ac444be24f71d08b99974c2f4b96e5')
     version('0.3-2', '4ce28f46df68fbb73905711ba2416fac')
 
-    depends_on('r@3.4.0:3.4.9')
+    depends_on('r@2.10:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-som/package.py]: `type=('build', 'run')`

File: var/spack/repos/builtin/packages/r-chemometrics/package.py [hunk 7]
```diff
@@ -39,7 +39,7 @@ class RChemometrics(RPackage):
     version('1.3.8', '7fad828bd094b5485fbf20bdf7d3d0d1')
     version('1.3.7', 'a9e2f32efb1545421dd96185fd849184')
 
-    depends_on('r@3.4.0:3.4.9')
+    depends_on('r@2.10:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-chemometrics/package.py]: `type=('build', 'run')`

File: var/spack/repos/builtin/packages/r-fnn/package.py [hunk 8]
```diff
@@ -40,6 +40,6 @@ class RFnn(RPackage):
     version('0.6-3', 'f0f0184e50f9f30a36ed5cff24d6cff2')
     version('0.6-2', '20648ba934ea32b1b00dafb75e1a830c')
 
-    depends_on('r@3.4.0:3.4.9')
+    depends_on('r@3.0.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-fnn/package.py]: `type=('build', 'run')`

**Comment by tgamblin** [comment]
Why is this called Tanim?

**Comment by mmpozulp** [comment]
@adamjstewart did I address all of your feedback? I believe that I made all the changes that you requested.

**Pull request merged**